android/src/main/java/com/learnium/RNDeviceInfo/resolver/DeviceTypeResolver.java

```diff
@@ -66,19 +67,29 @@ private DeviceType getDeviceTypeFromPhysicalSize() {
       return DeviceType.UNKNOWN;
     }
 
-    // Get display metrics to see if we can differentiate handsets and tablets.
-    // NOTE: for API level 16 the metrics will exclude window decor.
+    double widthInches;
+    double heightInches;
     DisplayMetrics metrics = new DisplayMetrics();
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
-      windowManager.getDefaultDisplay().getRealMetrics(metrics);
+    if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.R) {
+      // Get display metrics to see if we can differentiate handsets and tablets.
+      // NOTE: for API level 16 the metrics will exclude window decor.
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+        windowManager.getDefaultDisplay().getRealMetrics(metrics);
+      } else {
+        windowManager.getDefaultDisplay().getMetrics(metrics);
+      }
+      // Calculate physical size.
+      widthInches = metrics.widthPixels / (double) metrics.xdpi;
+      heightInches = metrics.heightPixels / (double) metrics.ydpi;
     } else {
-      windowManager.getDefaultDisplay().getMetrics(metrics);
+      final WindowMetrics windowMetrics = windowManager.getCurrentWindowMetrics();
+      metrics = context.getApplicationContext().getResources().getDisplayMetrics();
+
+      widthInches = windowMetrics.getBounds().width() / (double) metrics.xdpi;
+      heightInches = windowMetrics.getBounds().height() / (double) metrics.ydpi;
     }
 
-    // Calculate physical size.
-    double widthInches = metrics.widthPixels / (double) metrics.xdpi;
-    double heightInches = metrics.heightPixels / (double) metrics.ydpi;
-    double diagonalSizeInches = Math.sqrt(Math.pow(widthInches, 2) + Math.pow(heightInches, 2));
+    double diagonalSizeInches = Math.floor(Math.sqrt(Math.pow(widthInches, 2) + Math.pow(heightInches, 2)));
```

Comment:
Hi there!
I like the change to remove the deprecation by using new APIs, however, you've altered the implementation here with the addition of Math.floor, and indeed some of the breakpoint tests below are in non-integer ranges, so it's possible this will return a different result now if the handset has an actual diagonal width of 6.91 inches - now that will be a handset instead of a tablet as you'll round 6.91 down to 6.0 and it will fall into a different bracket of conditionals

```suggestion
    double diagonalSizeInches = Math.sqrt(Math.pow(widthInches, 2) + Math.pow(heightInches, 2));
```